test/e2e/broker_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 The Knative Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+// +build !serving
+
+package e2e
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+
+	"knative.dev/client/lib/test"
+	"knative.dev/client/pkg/util"
+)
+
+func TestBroker(t *testing.T) {
+	t.Parallel()
+	it, err := test.NewKnTest()
+	assert.NilError(t, err)
+	defer func() {
+		assert.NilError(t, it.Teardown())
+	}()
+
+	r := test.NewKnRunResultCollector(t, it)
+	defer r.DumpIfFailed()
+
+	t.Log("create broker, list and describe it")
+	brokerCreate(r, "foo1")
+	verifyBrokerList(r, "foo1")
+	verifyBrokerListOutputName(r, "foo1")
+	verifyBrokerDescribe(r, "foo1")
+	brokerDelete(r, "foo1")
+
+	t.Log("create broker and delete it")
+	brokerCreate(r, "foo2")
+	verifyBrokerList(r, "foo2")
+	brokerDelete(r, "foo2")
+	verifyBrokerNotfound(r, "foo2")
```

Comment:
@dsimansk : seems like the e2e failure 
```
    TestBroker: result_collector.go:84: ERROR: Error expected but no error happened
```
[log](https://prow.knative.dev/view/gcs/knative-prow/pr-logs/pull/knative_client/894/pull-knative-client-integration-tests-latest-release/1274800525564448769#1:build-log.txt%3A678)

is the scenario where the resource being deleted is described before delete completes?

Reply:
Per our chat discussion, there seems to be race condition between actual completion of `broker delete` operation out `verifyBrokerNotfound`. I've addes sync delete in commit https://github.com/knative/client/pull/894/commits/e7b526144d79b24b9c45e6ab36b77da647505105. Lets see if it helps.